### ISSUE TITLE
Lost EDDI TTS audio

### ISSUE THREAD

**Issue by youloveben** [EDCD/EDDI#356]
## EDDI version in which issue found
[2.4.6-b2]

## Steps to reproduce
1. [Been using VA and EDDI plugin for a year, but recently EDDI stopped being audible. Still acts functional- no errors that I can discern, it appears to still pings EDSM and local logs etc and system TTS works just fine]
2. [Using Cereproc Caitlin voice, but even with system default voice I get no audio from EDDI]
3. [Always appears initialized in VA upon load]

## Expected
[TTS audio in general]

## Observed
[no audio]

## Investigation
[tried using non-beta version 2.4.4 and reinstalled Cereproc, VA, and EDDI in case something got corrupted along the way. VA works, Cereproc works, no EDDI audio though]

[eddi_issue.zip](https://github.com/EDCD/EDDI/files/1658226/eddi_issue.zip)

**Comment by youloveben** [comment]
...now I can't even get EDDI to open. What did I do?

**Comment by Tkael** [comment]
EDDI doesn't appear to have detected or logged any errors.
Is there anything else that you've installed / reinstalled recently? Until the most recent betas, EDDI would crash if Elite Dangerous hadn't been run on a machine first and the location where journal files are saved hadn't yet been created...

**Comment by youloveben** [comment]
Yeah, I didn't see any errors in the verbose log, which is why I stopped by to ask here. Other than re-installing 2.4.6-b2, nothing new. Elite has it's log in the right spot, and has run several times since reinstalling EDDI. Again- I've had EDDI run beautifully for a while... no idea why it won't interact with TTS (or even open at this point).

**Comment by Tkael** [comment]
Ok. Try the above steps... you may have a corrupted config file.

**Comment by youloveben** [comment]
Thanks. I did as you directed, and noticed that several files repopulated... but the EDDI UI still won't manifest. I don't even see it running in Task Manager. D:

**Comment by youloveben** [comment]
...which is weird because VA reports it as initialized. I'm noticing a slew of EDDI.exe appcrash reports in C:\ProgramData\Microsoft\Windows\WER\ReportQueue now.

**Comment by Tkael** [comment]
Ok. If those are recent that might give us a hint as to what is happening.
Have you tried running EDDI standalone, without VoiceAttack?

**Comment by youloveben** [comment]
Sure have; can't get it to open alone or via VA. That's a new, odd development. Earlier it would open, just not TTS audio any reports.

**Comment by youloveben** [comment]
Okay- using Revo Uninstaller, I've done a clean wipe of all EDDI files. Reinstalled, and now it opens. Back to the original problem- no TTS audio.

**Comment by Tkael** [comment]
Maybe your personality file has become corrupted? Did you cut and paste your personality folder to your desktop too?
If you did, I'm not sure what else you might change (or might have changed). Is anything new getting written to EDDI's logs?

**Comment by youloveben** [comment]
[eddi_issue.zip](https://github.com/EDCD/EDDI/files/1658630/eddi_issue.zip)

**Comment by youloveben** [comment]
I did create an alternate EDDI personality file, but it's hidden away (backed up for when we sort this out) in a .zip file on a separate drive so as not to complicate matters. If I go into system TTS, I can get the Cereproc voice to demo, but I can't figure out why EDDI can't access it anymore?

**Comment by youloveben** [comment]
![capture](https://user-images.githubusercontent.com/35619358/35315245-2d596216-0080-11e8-859a-2e6e095b42e6.PNG)

**Comment by Tkael** [comment]
Have you made any recent changes to your audio drivers?
EDDI simply outputs to default system audio so as long as EDDI's volume isn't zero and the system volume isn't zero you should be able to hear responses.
Does a simple script with just text (no functions, variables, etc.) still fail to play when tested? Or when phonetic speech is disabled?
If you select a different voice, does audio still fail to play? (Cereproc voices aren't always 100% conforming to published standards...)

**Comment by Tkael** [comment]
No app errors other than the expected errors when EDDI doesn't find config files and creates them.

**Comment by Tkael** [comment]
How recently did you update to version 2.4.6-b2 and start using Caitlin? Did this problem begin when both of those first became true?

**Comment by youloveben** [comment]
Yeah that's why I attached the screencap of the audio mixer- EDDI is not muted. 

Looking through driver files, nothing looks new/updated. I updated EDDI beta as they were released, so it's been a few weeks, at least.

Cereproc Caitlin was installed this past summer and worked until around Nov/Dec until I finally noticed that VA was working but I wasn't getting EDDI TTS anymore. 

I forgot to mention that I also use TTS4ED, but it never had any conflicts before (and still works fine turning NPC chatter into TTS). 

None of the EDDI Speech Responder "test" buttons give audio either; it really feels like EDDI is muted!

I see the CPU working when I hit the UI buttons, so I assume it's all calculating... but how to confirm the audio throughput?

**Comment by youloveben** [comment]
https://sites.google.com/site/tts4ed/

**Comment by youloveben** [comment]
...and yes, outside of EDDI, Caitlin speaks anything I throw at 'er!

**Comment by Tkael** [comment]
Ok. We made some adjustments to the way that EDDI handles Cereproc voices with beta 2.4.6-b2 (ironically enough, these changes address problems another user was reporting with Cereproc Heather).
If you uninstall EDDI 2.4.6-b2 and install either EDDI 2.4.5 or 2.4.6-b1, does that fix the issue?

**Comment by Tkael** [comment]
Ref. #295.

**Comment by Tkael** [comment]
Also, you haven't said yet whether the problem persists when you select a different voice?

**Comment by youloveben** [comment]
Sorry I forgot to mention that yes, I've tried TTS default, MSMark and MSZira with no audible results. 

Can also confirm that EDDI has no problems obtaining my EDSM log and other credentials... So 2.4.6-b2 definitely "works" for me but is otherwise acting muted. Just uninstalled, then installed 2.4.5 and no go- still mute! Gah!

**Comment by youloveben** [comment]
[eddi_issue.zip](https://github.com/EDCD/EDDI/files/1658691/eddi_issue.zip)

**Comment by Tkael** [comment]
Huh. I have to say that I'm pretty stumped. Sorry. :-(

**Comment by youloveben** [comment]
I did try disabling phonetic speech; it was one of the first steps I took. Man... I'm going to miss EDDI.

**Comment by Tkael** [comment]
Hi, were you able to find a resolution for this issue?

**Comment by youloveben** [comment]
I did, and as you probably figured... it was just something local.

I did a few random adjustments to the recording devices options in Win10
and suddenly EDDI started choochin' along like nothing ever happened! I
don't recall specifically what I changed (conference mode, maybe?) but it
was some dumb local setting all along. Sorry to bother you about it; it was
totally on me.

EDDI rocks and goes really well with VA, TTS4ED, and EDEngineer. MY
IMMERSION IS BACK!

/ben/   (❍ᴥ❍ʋ)
